staging/src/k8s.io/client-go/util/flowcontrol/backoff.go

```diff
@@ -138,7 +159,10 @@ func (p *Backoff) DeleteEntry(id string) {
 
 // Take a lock on *Backoff, before calling initEntryUnsafe
 func (p *Backoff) initEntryUnsafe(id string) *backoffEntry {
-	entry := &backoffEntry{backoff: p.defaultDuration}
+	entry := &backoffEntry{
+		backoff:    p.defaultDuration,
+		randomness: rand.New(rand.NewSource(p.randomSeedSource())), // Create a per-item seed.
```

Comment:
I am confused about why it constructs a `rand` per entry? Is it for locking?

If this is really necessary, I recommend having Backoff construct a `rand` and use that to generate the seeds--that is easier to reason about.

Like, I think the current code in this PR has the surprising property that perfectly synchronized clocks would cause clients failing simultaneously to fall into exactly the same random number sequence? Admittedly that's not currently super likely, but it's bad karma to leave things like that laying around.

Reply:
Yeah, I can't decide why this is necessary.

Reply:
This was a misunderstanding on my part about how seeds work.